### PR TITLE
Move log and error messages to centralized files

### PR DESCRIPTION
**Pull request by orenbm**
Connected to #1539 

We maintain our logs and errors in the centralized files `logs.rb` & `errors.rb`. 

We had some log messages written outside these files so we moved them there.

It is best to review this PR via the commits as they are split logically. We will squash-merge this PR when it's ready. 

This PR blocks [this issue](https://github.com/cyberark/conjur/issues/1539) as it cleans up the code a bit so that issue's PR will be easier to review.

### CODE REVIEW

File: app/domain/errors.rb [hunk 1]
```diff
@@ -171,10 +171,20 @@ module AuthnOidc
...
+          msg:  "Invalid or expired AWS headers: {0}",
           code: "CONJ00018E"
         )
 
+        VerificationError = ::Util::TrackableLogMessageClass.new(
+          msg:  "Verification of IAM identity failed with exception: {0-exception}",
+          code: "CONJ00063E"
+        )
+
+        VerificationErrorCode = ::Util::TrackableLogMessageClass.new(
```
Comment (Tovli) [inline app/domain/errors.rb:183]: I suggest you'll use more meaningful names, as our approach of separating the log text to a separate files means that the developer is not aware of what is exactly the text from first glance

**Comment by orenbm** [comment]
@Tovli regarding your comment:

> I would prefer to have explicit collection of info level logs

This PR is a preparation for another one where all the messages will change from DEBUG to INFO so we don't to add a separation of files in this one.

File: app/controllers/concerns/authorize_resource.rb [hunk 2]
```diff
@@ -21,7 +21,13 @@ def authorize_many(resources, privilege)
 
   def auth(user, privilege, resource)
     unless user.allowed_to?(privilege, resource)
-      logger.info "Current user '#{user.role_id}' is not permitted to '#{privilege}' resource '#{resource.resource_id}'"
+      logger.info(
```
Comment (nessiLahav) [inline app/controllers/concerns/authorize_resource.rb:24]: Why not using logger.error instead of info?

Comment (orenbm) [reply app/controllers/concerns/authorize_resource.rb:24]: According to Datadog's post about [managing Rails application logs](https://www.datadoghq.com/blog/managing-rails-application-logs/), error logs indicate `errors that cause an operation to fail (e.g., missing data or files) but not the application`. In this case the operation didn't fail and we got to the conclusion successfully. We want to notify the client that there was an error in his flow so it's in info

**Pull request merged**